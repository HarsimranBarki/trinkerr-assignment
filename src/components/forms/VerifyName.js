import { motion } from "framer-motion";
import React, { useState } from "react";
import toast, { Toaster } from "react-hot-toast";
import { useHistory } from "react-router-dom";

const VerifyName = () => {
  const [name, setName] = useState("");
  const toastGenerate = (message) => toast.error(message);
  const history = useHistory();

  // Regex For Formatting Input Should Be Number
  const onlyAplhabets = (e) => {
    e.preventDefault();
    let value = e.target.value.replace(/[^a-zA-Z, ]/, "");
    setName(value);
  };

  const validateInput = (e) => {
    e.preventDefault();
    if (name.length === "") return toastGenerate("Name can't be empty");
    history.push("/survey");
  };

  return (
    <motion.form
      className="mt-10 "
      onSubmit={(e) => validateInput(e)}
      initial={{ opacity: 0, x: -10 }}
      animate={{ opacity: 1, x: 0 }}
      exit={{ opacity: 0, x: 10 }}
    >
      <Toaster position="bottom-center" reverseOrder={false} />
      <input
        type="text"
        className="form-input px-4 py-2 border-gray-300 rounded mt-2 shadow-inner"
        placeholder="Enter Your Name"
        value={name}
        onChange={(e) => onlyAplhabets(e)}
        autoFocus
      />
      <br />
      <button
        type="submit"
        className="bg-indigo-600 text-indigo-50 px-5 py-2  font-medium rounded mt-5 hover:bg-indigo-700 transition"
      >
        Submit
      </button>
    </motion.form>
  );
};

export default VerifyName;
