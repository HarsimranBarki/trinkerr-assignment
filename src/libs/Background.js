const Background = ({ children }) => {
  return (
    // Remove transition-all to disable the background color transition.
    <div className="bg-white dark:bg-gray-800 dark:text-gray-100 h-screen transition-all">
      {children}
    </div>
  );
};
export default Background;
