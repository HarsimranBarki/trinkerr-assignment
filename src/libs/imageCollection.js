import one from "../images/one.png";
import two from "../images/two.png";
import three from "../images/three.png";
import four from "../images/four.png";
import five from "../images/five.png";

export const imageCollection = [
  {
    id: 1,
    name: "one",
    imageUrl: one,
    swiped: false,
  },
  {
    id: 2,
    name: "two",
    imageUrl: two,
    swiped: false,
  },
  {
    id: 3,
    name: "three",
    imageUrl: three,
    swiped: false,
  },
  {
    id: 4,
    name: "four",
    imageUrl: four,
    swiped: false,
  },
  {
    id: 5,
    name: "five",
    imageUrl: five,
    swiped: false,
  },
];
