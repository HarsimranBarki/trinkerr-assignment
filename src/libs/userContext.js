import React, { useMemo, useState } from "react";

export const UserContext = React.createContext(null);

export const UserProvider = ({ children }) => {
  const [user, setUser] = useState(
    JSON.parse(localStorage.getItem("user")) || null
  );
  const providerValue = useMemo(() => ({ user, setUser }), [user, setUser]);

  return (
    <UserContext.Provider value={providerValue}>
      {children}
    </UserContext.Provider>
  );
};
