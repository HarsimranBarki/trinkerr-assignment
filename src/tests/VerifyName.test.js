import { fireEvent, render, screen } from "@testing-library/react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { createMemoryHistory } from "history";
import React from "react";
import VerifyName from "../components/forms/VerifyName";
import { UserProvider } from "../libs/userContext";

Enzyme.configure({ adapter: new Adapter() });

describe("VerifyName", () => {
  const setSignupState = jest.fn();
  const history = createMemoryHistory();
  it("VerifyName: should render correctly", () => {
    const wrapper = shallow(
      <UserProvider>
        <VerifyName setSignupState={setSignupState} history={history} />
      </UserProvider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("VerifyName: should have correct props ( :name )", () => {
    render(
      <UserProvider>
        <VerifyName setSignupState={setSignupState} history={history} />
      </UserProvider>
    );

    const buttonPhone = screen.queryByTestId("button");
    const inputName = screen.queryByTestId("input");

    fireEvent.change(inputName, { target: { value: "James" } });
    fireEvent.click(buttonPhone);
    expect(inputName.value).toBe("James");
    let vals = inputName.value.length;
    expect(vals).not.toBeNull();
  });

  it("VerifyName: should not allow null values", () => {
    render(
      <UserProvider>
        <VerifyName setSignupState={setSignupState} history={history} />
      </UserProvider>
    );
    const buttonPhone = screen.queryByTestId("button");
    const inputName = screen.queryByTestId("input");

    inputName.value = "";
    fireEvent.click(buttonPhone);
    let vals = inputName.value.length;

    expect(vals).not.toBeNull();
  });
});
