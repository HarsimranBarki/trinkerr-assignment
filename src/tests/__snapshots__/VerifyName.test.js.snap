// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyName VerifyName: should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": null,
    }
  }
>
  <VerifyName
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "wtt0g6",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "wtt0g6",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    setSignupState={[MockFunction]}
  />
</ContextProvider>
`;
